repos/system_upgrade/el7toel8/actors/removeoldpammodulescheck/actor.py

```diff
@@ -0,0 +1,101 @@
+from leapp.actors import Actor
+from leapp.dialogs import Dialog
+from leapp.dialogs.components import BooleanComponent
+from leapp.libraries.common.reporting import report_with_remediation
+from leapp.models import RemovedPAMModules
+from leapp.reporting import Report
+from leapp.tags import IPUWorkflowTag, ChecksPhaseTag
+
+
+class RemoveOldPAMModulesCheck(Actor):
+    """
+    Check if it is all right to disable PAM modules that are not in RHEL-8.
+
+    If admin will refuse to disable these modules (pam_pkcs11 and pam_krb5),
+    upgrade will be stopped. Otherwise we would risk locking out the system
+    once these modules are removed.
+    """
+    name = 'removed_pam_modules_check'
+    consumes = (RemovedPAMModules,)
+    produces = (Report,)
+    tags = (IPUWorkflowTag, ChecksPhaseTag)
+    dialogs = (
+        Dialog(
+            scope='remove_old_pam_modules_check',
+            reason='Confirmation',
+            components=(
+                BooleanComponent(
+                    key='confirm',
+                    label='Disable pam_pkcs11 module in PAM configuration? '
```

Comment:
Probably you should put information about 'yes' option here, as currently the message shows option for 'no' only. Might be confusing as the operator doesn't know what to put ('y' or 'yes')

Reply:
currently, it is not possible to use dialogs as interactive upgrades are no go now.

Reply:
let's discuss that next week, maybe I am missing something. But would like to stay clear: When upgrade is not possible to do without interactivity, it's no-go for many users. When there will be dialogs, there has to be way to skip any interactions and use defaults for example (and that should be default)

Reply:
I was under the impression that it some answer file and unattended upgrade is already supported by the leapp tool. If it is not the case and it won't be, I agree that running the upgrade in non-interactive way has precedence. However, in this case customers must have the option to opt out from the authselect actors and it should be thoroughly documented.

Reply:
> Probably you should put information about 'yes' option here, as currently the message shows option for 'no' only. Might be confusing as the operator doesn't know what to put ('y' or 'yes')

This should be handled by the leapp framework. I'm just using BooleanComponent with default=False which is represented by leapp as `Question? [no]` instead of 'Question? [yes/no] [default: no]' or something similar.